lib/system/io.nim

```diff
@@ -745,8 +745,10 @@ proc getFilePos*(f: File): int64 {.benign.} =
   result = c_ftell(f)
   if result < 0: raiseEIO("cannot retrieve file position")
 
-proc getFileSize*(f: File): int64 {.tags: [ReadIOEffect], benign.} =
+proc getFileSize*(f: File): int64 {.tags: [ReadIOEffect], benign, deprecated: "use `os.getFileSize`".} =
   ## retrieves the file size (in bytes) of `f`.
+  ## Caution: this uses `c_fseek(f, 0, SEEK_END)`, which is undefined behavior
+  ## on some systems, and can overflow for large files on others.
```

Comment:
Who claims it's UB? Why would ``SEEK_END`` exist if it's UB?

Reply:
see updated PR message